src/Microsoft.AspNet.Mvc.ModelBinding/BinderMetadata/BindingSource.cs

```diff
@@ -0,0 +1,171 @@
+            isValueProvider: false,
+            isUserInput: true);
+
+        /// <summary>
+        /// A <see cref="ApiParameterSource"/> for a custom model binder (unknown data source).
+        /// </summary>
+        public static readonly BindingSource Custom = new BindingSource(
+            "Custom",
+            Resources.BindingSource_Custom,
+            isValueProvider: false,
+            isUserInput: true);
+
+        /// <summary>
+        /// A <see cref="ApiParameterSource"/> for the request form-data.
+        /// </summary>
+        public static readonly BindingSource Form = new BindingSource(
+            "Form",
+            Resources.BindingSource_Form,
+            isValueProvider: true,
+            isUserInput: true);
+
+        /// <summary>
+        /// A <see cref="BindingSource"/> for the request headers.
+        /// </summary>
+        public static readonly BindingSource Header = new BindingSource(
+            "Header",
+            Resources.BindingSource_Header,
+            isValueProvider: false,
+            isUserInput: true);
+
+        /// <summary>
+        /// A <see cref="BindingSource"/> for model binding. Includes form-data, query-string
+        /// and route data from the request.
+        /// </summary>
+        public static readonly BindingSource ModelBinding = new BindingSource(
+            "ModelBinding",
+            Resources.BindingSource_ModelBinding,
+            isValueProvider: true,
+            isUserInput: true);
+
+        /// <summary>
+        /// An <see cref="ApiParameterSource"/> for the request url path.
+        /// </summary>
+        public static readonly BindingSource Path = new BindingSource(
+            "Path",
+            Resources.BindingSource_Path,
+            isValueProvider: true,
+            isUserInput: true);
+
+        /// <summary>
+        /// An <see cref="ApiParameterSource"/> for the request query-string.
+        /// </summary>
+        public static readonly BindingSource Query = new BindingSource(
+            "Query",
+            Resources.BindingSource_Query,
+            isValueProvider: true,
+            isUserInput: true);
+
+        /// <summary>
+        /// A <see cref="BindingSource"/> for request services.
+        /// </summary>
+        public static readonly BindingSource Services = new BindingSource(
+            "Services",
+            Resources.BindingSource_Services,
+            isValueProvider: false,
+            isUserInput: false);
+
+        public BindingSource([NotNull] string id, string displayName, bool isValueProvider, bool isUserInput)
+        {
+            Id = id;
+            DisplayName = displayName;
+            IsValueProvider = isValueProvider;
+            IsUserInput = isUserInput;
+        }
+
+        public string DisplayName { get; }
+
+        /// <summary>
+        /// Gets an 
+        /// </summary>
+        public string Id { get; }
+
+        public bool IsValueProvider { get; }
```

Comment:
After discussion this will just be `IsGreedy`. We're not splitting the semantics into two properties.